### PR TITLE
EXSWCPHIPT-113 - Unit tests for hipArray3DCreate for default and surface arrays

### PR DESCRIPTION
**Pull request by FMarno**
PR is stacked on #2713

### CODE REVIEW

File: tests/catch/unit/memory/DriverContext.hh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+THE SOFTWARE.
+*/
+
+#pragma once
+
+#include <hip_test_context.hh>
+
+class DriverContext {
+ private:
+#if HT_NVIDIA
```
Comment (chrispaquot) [inline tests/catch/unit/memory/DriverContext.hh]: You don't really have to protect this

File: tests/catch/unit/memory/DriverContext.cc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "DriverContext.hh"
+#include <hip_test_common.hh>
+
+DriverContext::DriverContext() {
+#if HT_NVIDIA
```
Comment (chrispaquot) [inline tests/catch/unit/memory/DriverContext.cc]: Do we really need to protect for HT_NVIDIA here?

Comment (FMarno) [reply tests/catch/unit/memory/DriverContext.cc]: The tests pass either way. Would you prefer I remove it then?

File: tests/catch/unit/memory/DriverContext.hh (hunk 1 above)
Comment (FMarno) [reply tests/catch/unit/memory/DriverContext.hh]: If I remove the implementation code then the compiler complains about unused variables.

File: tests/catch/unit/memory/DriverContext.cc (hunk 2 above)
Comment (chrispaquot) [reply tests/catch/unit/memory/DriverContext.cc]: yes

File: tests/catch/unit/memory/DriverContext.hh (hunk 1 above)
Comment (chrispaquot) [reply tests/catch/unit/memory/DriverContext.hh]: umm... ok, this is fine then.

File: tests/catch/unit/memory/DriverContext.cc (hunk 2 above)
Comment (FMarno) [reply tests/catch/unit/memory/DriverContext.cc]: I've done that now.

File: tests/catch/unit/memory/DriverContext.hh (hunk 1 above)
Comment (FMarno) [reply tests/catch/unit/memory/DriverContext.hh]: Since I removed the other preprocessor macros this has also been removed now.

**Review by chrispaquot — DISMISSED** [review]

File: tests/catch/unit/memory/hipMalloc3DArray.cc [hunk 3]
```diff
@@ -176,20 +126,145 @@ TEST_CASE("Unit_hipMalloc3DArray_MultiThread") {
...
+  const auto avail = getFreeMem();
 
   if (pavail != avail) {
     WARN("Memory leak of hipMalloc3D API in multithreaded scenario");
     REQUIRE(false);
   }
 }
 
+namespace {
+void checkArrayIsExpected(hipArray_t array, const hipChannelFormatDesc& expected_desc,
```
Comment (chrispaquot) [inline tests/catch/unit/memory/hipMalloc3DArray.cc:149]: What's the point of this function?

Comment (FMarno) [reply tests/catch/unit/memory/hipMalloc3DArray.cc:149]: like #2713, I have updated it do actually do something for the NVIDIA backend.

**Review by chrispaquot — DISMISSED** [review]

**Review by mangupta — APPROVED** [review]
Adding back approval

**Pull request merged**